qutip/solver/parallel.py

```diff
@@ -145,39 +146,41 @@ def parallel_map(task, values, task_args=None, task_kwargs=None,
     if task_kwargs is None:
         task_kwargs = {}
     map_kw = _read_map_kw(map_kw)
-    os.environ['QUTIP_IN_PARALLEL'] = 'TRUE'
     end_time = map_kw['timeout'] + time.time()
     job_time = map_kw['job_timeout']
 
     progress_bar = progess_bars[progress_bar]()
     progress_bar.start(len(values), **progress_bar_kwargs)
 
-    results = []
-    try:
-        pool = Pool(processes=map_kw['num_cpus'])
-
-        async_res = [pool.apply_async(task, (value,) + task_args, task_kwargs)
-                     for value in values]
-
-        for job in async_res:
-            remaining_time = min(end_time - time.time(), job_time)
-            result = job.get(remaining_time)
-            if reduce_func is not None:
-                reduce_func(result)
-            else:
-                results.append(result)
-            progress_bar.update()
-
-    except KeyboardInterrupt as e:
-        raise e
-
-    except multiprocessing.TimeoutError:
-        pass
+    if sys.version_info >= (3, 7):
+        # ProcessPoolExecutor only supports mp_context from 3.7 onwards
+        ctx_kw = {"mp_context": mp_context}
+    else:
+        ctx_kw = {}
 
+    os.environ['QUTIP_IN_PARALLEL'] = 'TRUE'
+    try:
```

Comment:
I hadn't realized the intention of catching the error. Let me see if I can fix things up so that results are retained properly.

Maybe I can also fix `ProcessPoolExecutor` waiting too long.